### PR TITLE
Python 2 / 3 compatible imports.

### PR DESCRIPTION
**Pull request by rudaporto**

### CODE REVIEW

**Comment by rudaporto** [comment]
@pbauer can you help me with this review or should I ask someone else?

**Review by tomgross — COMMENTED** [review]
Thanks!
Is six a dependency in setup.py already?
A changelog entry is missing.

**Comment by rudaporto** [comment]
@tomgross I update with more compatible changes.
- remove some unused imports of urllib.quote_plus
- change to use built-in set class when sets (deprecated) module is not available
- add six.text_type instead of unicode

Still, there is some unused imports but I did not removed them at this stage.

**Comment by rudaporto** [comment]
@tomgross should I change travis.yaml to use tox run the build with python 2.7, 3.4, 3.5 and 3.6?

File: Products/PluggableAuthService/plugins/ScriptablePlugin.py [hunk 1]
```diff
@@ -15,8 +15,11 @@
 
 $Id$
 """
-from sets import Set
-from urllib import quote_plus
+try:
+    from sets import Set as set
+except ImportError:
+    pass
+
```
Comment (tseaver) [inline Products/PluggableAuthService/plugins/ScriptablePlugin.py]: Just remove the import and *always* use the built-in `set` type.

Comment (rudaporto) [reply Products/PluggableAuthService/plugins/ScriptablePlugin.py]: Done.

**Comment by tomgross** [comment]
@rudaporto This package depends on GenericSetup (Python 3 upgrade is WIP) and ZServer (This package will probably never see Python 3)

To get an overview what is achieved it is probably a good idea to have them with `allow_failures` option on.
I don't know, if there is an official statement on Python 3 subversion compatibility but I'd focus on 3.5 and 3.6.

**Comment by tomgross** [comment]
Maybe it is a good idea to have coveralls run too to see where the blind spots are.

**Comment by tomgross** [comment]
Apart from that this LGTM.

**Comment by tseaver** [comment]
We should make the dependency on `ZServer` optional.  The dependency is there to support detection of an `FTPRequest`, but if `ZServer` is not installed, we can just create a no-op class, e.g.:

```python
# in Products/PluggableAuthService/plugins/RequestTypeSniffer.py
try:
    from ZServer.FTPRequest import FTPReques
except ImportError:
    class FTPRequest(object):
        """Dummy for missing ZServer's 'FTPRequest'."""
```

**Comment by icemac** [comment]
@tseaver +1 to make ZServer optional, some weeks ago I created #8 to address this issue. Maybe it can be done in another PR.

**Review by icemac — APPROVED** [review]
LGTM

**Comment by icemac** [comment]
What has still to be done to get this PR merged?

**Pull request merged**